boa/src/builtins/typed_arrays/typed_array.rs

```diff
@@ -51,13 +51,42 @@ impl TypedArray {
         constructor.into()
     }
 
-    pub(crate) fn of(this: &Value, arguments: &[Value], context: &mut Context) -> Result<Value> {
-        let length: Value = arguments.len().into();
+    fn construct_species(
+        this: &Value,
+        arguments: &[Value],
+        context: &mut Context,
+    ) -> Result<Value> {
+        let length = arguments
+            .get(0)
+            .map(|v| v.get_field("length", context))
+            .transpose()?
+            .unwrap_or(0.into());
 
-        let constructed_value = this
+        let species = this
             .as_object()
             .ok_or_else(|| -> Value { "Not a constructor".into() })?
```

Comment:
Shouldn't this use `context.throw_type_error()` instead?

Reply:
It probably should, yeah good idea.